File: Clg_Placement/C++/Arrays/_47.ProductOfMaxiInFirstArrayAndMiniInSecondArray.cpp
#include <climits>

class Solution {
public:
    long long find_multiplication(int a[], int b[], int n, int m) {
        long long maxi = LLONG_MIN;
        long long mini = LLONG_MAX;
        
        for (int i = 0; i < n; i++)
            maxi = max(maxi, static_cast<long long>(a[i]));

        for (int i = 0; i < m; i++)
            mini = min(mini, static_cast<long long>(b[i]));

        return maxi * mini;
    }
};

/*
By using long long for both maxi and mini, you minimize the risk of overflow when calculating 
the multiplication. Also, note the use of static_cast<long long>(...) to explicitly convert array 
elements to long long to avoid potential issues when comparing and assigning values.
*/

File: Clg_Placement/Java/_47.ProductOfMaxiInFirstArrayAndMiniInSecondArray.java
class Solution{
    
    // Function for finding maximum and value pair
    public static long find_multiplication (int arr[], int brr[], int n, int m) {
        long maxi = Long.MIN_VALUE;
        long mini = Long.MAX_VALUE;
        for(int i=0;i<n;i++){
            maxi = Math.max(maxi,arr[i]);
        }
        for(int i=0;i<m;i++){
            mini = Math.min(mini,brr[i]);
        }
        return mini*maxi;
    }
}
